Dockerfile

```
@@ -1,167 +1,131 @@
-FROM ubuntu:20.04 as builder
-
-# Install general dependencies
-RUN apt update && \
-    DEBIAN_FRONTEND=noninteractive apt install -y \
-        # Needed for pre-commit in next build stage
-        git \
-        libpcre2-8-0 \
-        # Builder deps
-        unzip \
-        software-properties-common \
-        curl \
-        python3 \
-        python3-pip && \
-    # Upgrade pip for be able get latest Checkov
-    python3 -m pip install --upgrade pip && \
-    # Cleanup
-    rm -rf /var/lib/apt/lists/*
+ARG base_image_version=3.9.7-alpine3.14
+FROM python:${base_image_version} as builder
+
+SHELL ["/bin/ash", "-eo", "pipefail", "-c"]
 
 ARG PRE_COMMIT_VERSION=${PRE_COMMIT_VERSION:-latest}
 ARG TERRAFORM_VERSION=${TERRAFORM_VERSION:-latest}
-
-# Install pre-commit
-RUN [ ${PRE_COMMIT_VERSION} = "latest" ] && pip3 install --no-cache-dir pre-commit \
-    || pip3 install --no-cache-dir pre-commit==${PRE_COMMIT_VERSION}
-
-# Install terraform because pre-commit needs it
-RUN curl -fsSL https://apt.releases.hashicorp.com/gpg | apt-key add - && \
-    apt-add-repository "deb [arch=amd64] https://apt.releases.hashicorp.com $(lsb_release -cs) main" && \
-    apt update && \
-    ( \
-        [ "$TERRAFORM_VERSION" = "latest" ] && apt install -y terraform \
-        || apt install -y terraform=${TERRAFORM_VERSION} \
-    ) && \
-    # Cleanup
-    rm -rf /var/lib/apt/lists/*
-
-#
-# Install tools
-#
-WORKDIR /bin_dir
-
 ARG CHECKOV_VERSION=${CHECKOV_VERSION:-false}
 ARG TERRAFORM_DOCS_VERSION=${TERRAFORM_DOCS_VERSION:-false}
 ARG TERRAGRUNT_VERSION=${TERRAGRUNT_VERSION:-false}
 ARG TERRASCAN_VERSION=${TERRASCAN_VERSION:-false}
 ARG TFLINT_VERSION=${TFLINT_VERSION:-false}
 ARG TFSEC_VERSION=${TFSEC_VERSION:-false}
+ARG INSTALL_ALL=${INSTALL_ALL:-false}
 
+# hadolint ignore=DL3018
+RUN apk add --no-cache jq curl unzip && rm -rf "/var/cache/apk/*"
```

`--no-cache` == `rm -rf "/var/cache/apk/*"`

```suggestion
RUN apk add --no-cache jq curl unzip
```